lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -301,5 +303,103 @@ def self.cleanup(method_pid, threads)
+          rest_ws = {:workspace => obj.workspace.hash_workspace,
+                     :inputs    => inputs,
+                     :current   => current_info(obj.workspace),
+                     :state_var => state_var_hash}.to_yaml
+          open(input_file, 'w') { |f| f.puts rest_ws }
+          File.delete(output_file) if File.exist?(output_file)
+
+          preamble = rest_method_preamble(input_file, output_file, miq_method_token(obj))
+          obj.workspace.num_rest_methods += 1
+          $miq_ae_logger.info("<AEMethod [#{aem.fqname}]> Starting ")
+          rc, msg, stderr = run_ruby_method(aem.data, preamble)
+          $miq_ae_logger.info("<AEMethod [#{aem.fqname}]> Ending")
+          process_ruby_method_results(rc, msg, stderr)
+          update_workspace(obj, output_file)
+        ensure
+          obj.workspace.num_rest_methods -= 1
+          #File.delete(input_file) if File.exist?(input_file)
+          #File.delete(output_file) if File.exist?(output_file)
+        end
+      end
+    end
+
+    REST_RUBY_METHOD_PREAMBLE = <<-RUBY
+begin
+  require 'date'
+  require 'miq_ae_method_wrapper'
+  $evm = MiqAeMethodWrapper.new
+end
+
+begin
+RUBY
+    def self.rest_method_preamble(input_file, output_file, token)
+      preamble = "$:.unshift('#{Rails.root.join("lib/miq_automation_engine/rest")}')\n"
+      preamble << "MIQ_INPUT_FILE = '#{input_file}'\n"
+      preamble << "MIQ_OUTPUT_FILE = '#{output_file}'\n"
+      preamble << "MIQ_METHOD_TOKEN = '#{token}'\n"
+      preamble << REST_RUBY_METHOD_PREAMBLE
+      $miq_ae_logger.info("<AEMethod> Preamble")
+      $miq_ae_logger.info(preamble)
+      preamble
+    end
+
+    def self.miq_method_token(obj)
+      ae_user = obj.workspace.ae_user
+      tm = TokenManager.new("api")
+      tm.gen_token("api", :userid => ae_user.userid)
+    end
+
+    def self.update_workspace(obj, output_file)
+      hash = YAML.load_file(output_file)
+      update_object(obj.workspace, hash['workspace'])
+      hash['state_var'].each { |k, v| obj.workspace.presist_state_hash[k] = v }
+    end
+
+    def self.update_object(workspace, obj_hash)
+      path = "/#{obj_hash['namespace']}/#{obj_hash['class']}/#{obj_hash['instance']}"
+
+      obj = find_object(workspace.root, path)
+      $miq_ae_logger.error("Object #{path} not found in workspace") unless obj
+      raise MiqAeException::Error, "object not found #{path}" unless obj
+
+      update_obj_attributes(obj, obj_hash['attributes']) if obj_hash['attributes']
+      update_obj_references(obj, obj_hash['references']) if obj_hash['references']
+
+      obj_hash['MiqAeObject'].each { |hash| update_object(workspace, hash) } if obj_hash['MiqAeObject']
+    end
+
+    def self.find_object(obj, object_name)
+      return obj if obj.object_name == object_name
+      obj.children.each do |child|
+        if found = find_object(child, object_name)
+          break found
```

Comment:
You can use `detect` here: `obj.children.detect { |child| find_object(child, object_name) }